commands/list.go

```diff
@@ -50,117 +77,144 @@ List requires a subcommand, e.g. ` + "`hugo list drafts`.",
+					return newSystemError("Error building sites", err)
 				}
 
 				writer := csv.NewWriter(os.Stdout)
 				defer writer.Flush()
 
 				for _, p := range sites.Pages() {
 					if resource.IsFuture(p) {
-						err := writer.Write([]string{filepath.Join(p.File().Dir(), p.File().LogicalName()), p.PublishDate().Format(time.RFC3339)})
+						err := writer.Write([]string{
+							strings.TrimPrefix(p.File().Filename(), sites.WorkingDir+string(os.PathSeparator)),
+							p.PublishDate().Format(time.RFC3339),
+						})
 						if err != nil {
 							return newSystemError("Error writing future posts to stdout", err)
 						}
 					}
 				}
 
 				return nil
-
 			},
 		},
 		&cobra.Command{
 			Use:   "expired",
 			Short: "List all posts already expired",
-			Long: `List all of the posts in your content directory which has already
-expired.`,
+			Long:  `List all of the posts in your content directory which has already expired.`,
 			RunE: func(cmd *cobra.Command, args []string) error {
-				cfgInit := func(c *commandeer) error {
-					c.Set("buildExpired", true)
-					return nil
-				}
-				c, err := initializeConfig(true, false, &cc.hugoBuilderCommon, cc, cfgInit)
-				if err != nil {
-					return err
-				}
-
-				sites, err := hugolib.NewHugoSites(*c.DepsCfg)
+				sites, err := cc.buildSites(map[string]interface{}{"buildExpired": true})
 
 				if err != nil {
-					return newSystemError("Error creating sites", err)
-				}
-
-				if err := sites.Build(hugolib.BuildCfg{SkipRender: true}); err != nil {
-					return newSystemError("Error Processing Source Content", err)
+					return newSystemError("Error building sites", err)
 				}
 
 				writer := csv.NewWriter(os.Stdout)
 				defer writer.Flush()
 
 				for _, p := range sites.Pages() {
 					if resource.IsExpired(p) {
-						err := writer.Write([]string{filepath.Join(p.File().Dir(), p.File().LogicalName()), p.ExpiryDate().Format(time.RFC3339)})
+						err := writer.Write([]string{
+							strings.TrimPrefix(p.File().Filename(), sites.WorkingDir+string(os.PathSeparator)),
+							p.ExpiryDate().Format(time.RFC3339),
+						})
 						if err != nil {
 							return newSystemError("Error writing expired posts to stdout", err)
 						}
-
 					}
 				}
 
 				return nil
+			},
+		},
+		&cobra.Command{
+			Use:   "all",
+			Short: "List all posts",
+			Long:  `List all of the posts in your content directory, include drafts, future and expired pages.`,
+			RunE: func(cmd *cobra.Command, args []string) error {
+				sites, err := cc.buildSites(map[string]interface{}{
+					"buildExpired": true,
+					"buildDrafts":  true,
+					"buildFuture":  true,
+				})
+
+				if err != nil {
+					return newSystemError("Error building sites", err)
+				}
+
+				writer := csv.NewWriter(os.Stdout)
+				defer writer.Flush()
+
+				writer.Write([]string{
+					"path",
+					"slug",
+					"title",
+					"date",
+					"expiry_date",
+					"publish_date",
+					"draft",
+					"url",
```

Comment:
This overall PR looks good, but I think we need to stick to whatever fields we had in the existing list commands. It may be tempting to add more, but you will quickly get into questions about what URL / word count, etc. to use (re. output formats).

Reply:
Did I understand correctly, you think that additional fields like "url, word count, tags, categories" is unnecessary?

Reply:
I’m saying that there may be more than one URL, wordcount etc. for a page.

Reply:
But for most cases, I guess it will give the URL to the HTML version, which is probably what you want.

But:

* drop tags and categories (that is just relevant for people with tags and categories)
* drop word_count (for reason mentioned above)
* use the same naming standard for fields as in the struct; e.g. permalink, publishDate etc.

Reply:
> I’m saying that there may be more than one URL, wordcount etc. for a page.

But how? Can you give example?

> drop tags and categories, drop word_count, use the same naming standard

Done.